### ISSUE TITLE
Requesting for sharing preprocessing script

### ISSUE THREAD

**Issue by svjan5** [RingBDStack/dependency-based-w2v#1]
Hi,
I wanted to know is there any script for generating `depsl.txt` and `weightcn.txt` from raw text. If possible please share the same. 

Shikhar

**Comment by lreaderl** [comment]
Well, I'll upload them later.

**Comment by manikbhandari** [comment]
Hi,
I am also struggling with the same issue. Please try to upload it asap.

Thanks,
Manik

**Comment by lreaderl** [comment]
I've added, please check. If there's any problem, please tell me.

**Comment by manikbhandari** [comment]
Thanks a lot!

**Comment by manikbhandari** [comment]
Could you also share the script to create vocab file?

Thanks

**Comment by lreaderl** [comment]
I‘ve added vocab.py and updated count.py in step 4, please check.

**Comment by svjan5** [comment]
Hi! thanks for sharing the required preprocessing scripts. 
I tried to follow all the steps mentioned in the preprocessing folder but my program crashes with `segmentation fault` after some time (after ~2% progress). One difference which I noticed in my output of step3 is that it contains several `-1` although the demo file didn't have any. Do you have any suggestions for preventing segmentation error?

Thanks in advance
Shikhar

**Comment by manikbhandari** [comment]
Hi,
I am also facing a segmentation fault after around 1% progress. What could be the possible reasons for this?

**Comment by lllchenlll** [comment]
The appearance of '-1' means that the corresponding dependency is not found in the dependency list.
You can try to randomize a dependency (not too many '-1') or regenerate the depsl.txt

A segmentation fault may be caused by too many threads, you can try to reduce the number of threads appropriately.

**Comment by lreaderl** [comment]
Sorry, a lot busy.
The most possible reason of segmentation fault is from` -1` of the output of step3. Basically, you can replace these ` -1` with `0`. However,  ` -1` appears means that there are some relations in the output of step2 that never appear in `depsl.txt` in step3. I suggest you generate that file by yourself. And you can try to reduce the number of threads, large number of threads will lead to segmentation fault, too.

**Comment by svjan5** [comment]
Thank you for your help, it worked!

**Comment by manikbhandari** [comment]
Hi,
Thanks for your help in resolving the segmentation fault. After following the preprocessing steps, this is how my [weightcn](https://gist.github.com/svjan5/eb8ee3fe63b862ad5ec829e8a69d3c90#file-weightcn-txt), [vocab](https://gist.github.com/svjan5/eb8ee3fe63b862ad5ec829e8a69d3c90#file-vocab-txt) and [input data](https://gist.github.com/svjan5/eb8ee3fe63b862ad5ec829e8a69d3c90#file-mdeps-txt) look like. Unfortunately I'm getting very poor results on intrinsic evaluation tasks that you mentioned in the paper. Could you have a look please let me know if I'm making a mistake in preprocessing/training?

What could be other sources of poor results?

**Comment by lllchenlll** [comment]
Hi,
I don't know why I cannot access your files, however, I think these files should be fine.
I guess there are several reasons for your problem:
1) The order of dependencies. The corpus processing script we provided is a simplified version of the 3rd-order dependencies (there are average 5-8 words in target word's context), which far less than the whole dependency version or the traditional word2vec (there are average 10-16 words in target word's context when the -window = 8(defalut)), so the result will be reduced greatly.
2) The number of iterations. Since our code regards each line in corpus as a single training sample, the number of iterations for training needs to be set higher.
3) Hyperparameters. The hyperparameters in our model need to be adjusted according to the corpus. You can return them to 1 first, then gradually adjusted according to the results.
4) The corpus composition. Although we utilize many methods (e.g. sub-sampling) for solving the unbalance problem of dependency relationships, some dependencies are not trained sufficient.
5) The number of threads. Because the learning rate is shared by all threads, too many threads will lead to a lower results.
6) Initial randomization. The quality of the output results have a certain relationship with the initialization random, so there will be some fluctuations.
7) In addition, I don't know if the -1 you mentioned in the corpus is too much, I think this may be a reason.

I hope mentioned content will help you.

**Comment by prateeky2806** [comment]
Hey, 
I am also trying to train your model as I want to use these dependency based embeddings in my work as my problem might benefit from this initialization. I am also facing some issue during training. The _weightcn_ file for me seems to be the problem. 

If you can please share the trained embeddings(of dimension 100 and 300) used in the paper it would be really helpful. 
Thank you for actively helping people with issues.
